engine/access/subscription/base_tracker.go

```diff
@@ -0,0 +1,214 @@
+package subscription
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/common/rpc"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/state_synchronization"
+	"github.com/onflow/flow-go/module/state_synchronization/indexer"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+// GetStartHeightFunc is a function type for getting the start height.
+type GetStartHeightFunc func(context.Context, flow.Identifier, uint64) (uint64, error)
+
+// GetHighestHeight is a function type for getting the highest height.
+type GetHighestHeight func(flow.BlockStatus) (uint64, error)
+
+// StreamingData represents common streaming data configuration for access and state_stream handlers.
+type StreamingData struct {
+	MaxStreams  int32
+	StreamCount atomic.Int32
+}
+
+func NewStreamingData(maxStreams uint32) StreamingData {
+	return StreamingData{
+		MaxStreams:  int32(maxStreams),
+		StreamCount: atomic.Int32{},
+	}
+}
+
+// BaseTracker is an interface for a tracker that provides base GetStartHeight method related to both blocks and execution data tracking.
+type BaseTracker interface {
+	// GetStartHeight returns the start height to use when searching.
+	// Only one of startBlockID and startHeight may be set. Otherwise, an InvalidArgument error is returned.
+	// If a block is provided and does not exist, a NotFound error is returned.
+	// If neither startBlockID nor startHeight is provided, the latest sealed block is used.
+	//
+	// Returns expected errors:
+	// - codes.InvalidArgument: If blockStatus is flow.BlockStatusUnknown, or both startBlockID and startHeight are provided.
```

Comment:
This needs some updating, it doesn't work with block status anymore. 
Please check comment about my reasoning about sentinel errors in `engine/access/subscription/block_tracker.go`